cynthion/python/Makefile

```diff
@@ -62,3 +79,50 @@ assets/moondancer.bin:
 			python -m cynthion.gateware.$(basename $@).top --dry-run --output "assets/$$lp/$@"; \
 		fi; \
 	done
+
+
+# - environment checks --------------------------------------------------------
+
+detect-fpga-toolchain:
+ifeq (, $(shell which nextpnr-ecp5))
+	$(error "No FPGA toolchain detected. Please install: https://github.com/YosysHQ/oss-cad-suite-build/")
+endif
+	@echo "Using FPGA toolchain: $(shell dirname $(shell which nextpnr-ecp5))"
+
+detect-rust-toolchain:
+ifeq (, $(shell which rustup))
+	$(error "No Rust toolchain detected. Please install https://rustup.rs/")
+endif
+ifeq (, $(shell	rustup component list --installed|grep rust-std-riscv32imac-unknown-none-elf))
+	$(error "No riscv target support detected. Please run: 'rustup target add riscv32imac-unknown-none-elf'")
+endif
+ifeq (, $(shell find ~/.rustup/toolchains -name llvm-objcopy))
+	$(error "No llvm tools detected. Please run: 'rustup component add llvm-tools'")
+endif
+ifeq (, $(shell which cargo-objcopy))
+	$(error "No cargo-binutils detected. Please run: 'cargo install cargo-binutils'")
+endif
+	@echo "Using Rust toolchain: $(shell rustup show home) $(shell rustup show active-toolchain)"
+
+detect-luna-platform: detect-cynthion-cli
+ifeq ($(origin LUNA_PLATFORM), undefined)
+	@echo "LUNA_PLATFORM is not set, trying to autodetect it."
+	$(eval LUNA_PLATFORM := $(shell cynthion build --luna-platform 2> /dev/null))
+	@if [ ! -n "$(LUNA_PLATFORM)" ]; then \
+		echo "Detection failed" ; \
+		echo "" ; \
+		echo "Please ensure Cynthion is plugged in or set the LUNA_PLATFORM variable, for" ; \
+		echo "example:" ; \
+		echo "" ; \
+		echo " LUNA_PLATFORM=cynthion.gateware.platform:CynthionPlatformRev1D4 make assets" ; \
+		echo "" ; \
+		exit 1 ; \
+	fi
+endif
+	$(eval PLATFORM_NAME := $(shell echo $(LUNA_PLATFORM) | cut -d':' -f2))
+	@echo Using LUNA_PLATFORM: $(LUNA_PLATFORM)
+
+detect-cynthion-cli:
+ifeq (, $(shell which cynthion))
```

Comment:
A situation this does not handle is a previous version of `cynthion` installed that does not support `--luna-platform`. I think we can live with this for now, but maybe we should think about having a `make` target that installs in a `venv`? We do this in [cynthion-test](https://github.com/greatscottgadgets/cynthion-test/blob/main/Makefile).

Reply:
I've added a little check and a message so long:

```
"Unsupported cynthion command line tool detected. Please upgrade to the latest version."
```

Reply:
https://github.com/greatscottgadgets/cynthion/pull/166/commits/6f275a1ef29ba918fcf54e9644d14d829dac6d50